drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -0,0 +1,480 @@
+# -*- coding: utf-8 -*-
+#
+
+# Imports ###########################################################
+
+import json
+import webob
+import copy
+import urllib
+
+from xblock.core import XBlock
+from xblock.fields import Scope, String, Dict, Float, Boolean
+from xblock.fragment import Fragment
+from xblockutils.resources import ResourceLoader
+from xblockutils.settings import XBlockWithSettingsMixin, ThemableXBlockMixin
+
+from .utils import _  # pylint: disable=unused-import
+from .default_data import DEFAULT_DATA
+
+
+# Globals ###########################################################
+
+loader = ResourceLoader(__name__)
+
+
+# Classes ###########################################################
+
+@XBlock.wants('settings')
+class DragAndDropBlock(XBlock, XBlockWithSettingsMixin, ThemableXBlockMixin):
+    """
+    XBlock providing a Drag and Drop question
+    """
+    display_name = String(
+        display_name=_("Title"),
+        help=_("The title of the Drag and Drop that is displayed to the user"),
+        scope=Scope.settings,
+        default=_("Drag and Drop"),
+    )
+
+    show_title = Boolean(
+        display_name=_("Show title"),
+        help=_("Display the title to the user?"),
+        scope=Scope.settings,
+        default=True,
+    )
+
+    question_text = String(
+        display_name=_("Question text"),
+        help=_("The question text (and/or instructions) shown to the user"),
+        scope=Scope.settings,
+        default="",
+    )
+
+    show_question_header = Boolean(
+        display_name=_("Show \"Question\" heading"),
+        help=_("Display the \"Question\" heading above the question/instructions?"),
+        scope=Scope.settings,
+        default=True,
+    )
+
+    weight = Float(
+        display_name=_("Weight"),
+        help=_("This is the maximum score that the user receives when he/she successfully completes the problem"),
+        scope=Scope.settings,
+        default=1,
+    )
+
+    item_background_color = String(
+        display_name=_("Item background color"),
+        help=_(
+            "Background color to use for draggable items. "
+            "Defaults to color specified by current theme, or blue if no theme is set."
+        ),
+        scope=Scope.settings,
+        default="",
+    )
+
+    item_text_color = String(
+        display_name=_("Item text color"),
+        help=_(
+            "Text color to use for draggable items. "
+            "Defaults to color specified by current theme, or white if no theme is set."
+        ),
+        scope=Scope.settings,
+        default="",
+    )
+
+    data = Dict(
+        display_name=_("Drag and Drop"),
```

Comment:
@pdesjardins Course authors can not edit the `data` field directly. It stores information about a problem that is derived from the input that a course author provides via the interactive editor (in Studio). This string and the help string below are not displayed to users, so the information they contain is currently only useful for developers. I updated both of them to be more specific about the purpose of the field.